federated-learning-controller/internal/controller/manifests/client/manifestwork.yaml

```diff
@@ -47,4 +51,28 @@ spec:
                 - client # Maps to the entrypoint.sh logic
                 - --data-config={{ .ClientDataConfig }}
                 - --server-address={{ .ServerAddress }}
+                volumeMounts:
+                  - name: metric-data
+                    mountPath: /metrics
+              
+              {{- if .ObsSidecarImage }}
+              - name: obs-sidecar
+                image: {{ .ObsSidecarImage }}
+                imagePullPolicy: Always
+                args:
+                  - -metricfile=/metrics/metric.json
+                  - -endpoint=otel-collector.open-cluster-management-agent-addon.svc.cluster.local:4317
```

Comment:
Hard-coded OTel endpoint should be configurable. Consider adding an environment variable or using service discovery:

```yaml
- name: OTEL_ENDPOINT
  value: "otel-collector.open-cluster-management-agent-addon.svc.cluster.local:4317"
```

Then reference it in the args: `-endpoint=$(OTEL_ENDPOINT)`